### PR TITLE
composer arguments

### PR DESCRIPTION
**Pull request by ramunasd**
This adds additional option 'arguments' for composer. Sometimes it's very useful with commands init, require, create-project and etc.

### CODE REVIEW

**Comment by ramunasd** [comment]
thanks @resmo, but how can i know to which version this will be added?

**Comment by resmo** [comment]
set it to "2.0" for now

File: packaging/language/composer.py [hunk 1]
```diff
@@ -116,6 +132,8 @@ def parse_out(string):
     return re.sub("\s+", " ", string).strip()
 
 def has_changed(string):
+    if string == "":
```
Comment (resmo) [inline packaging/language/composer.py]: IMHO this is not needed, why did ou make it?

**Comment by robynbergeron** [comment]
Hi @ramunasd -- thanks for this PR. And thanks, @resmo for your quick review, as always :)

@resmo is one of the module maintainers, but adding the below just for process / clarity. @ramunasd -- I'm putting this in needs_revision, per @resmo's questions. :)

@resmo @dmtrs -- Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.
If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

File: packaging/language/composer.py (hunk 1 above)
Comment (ramunasd) [inline packaging/language/composer.py]: Sometimes composer returns nothing, but ansible thinks that something changed, because empty string doesn't have string 'Nothing to install...'.

Maybe this condition should be improved somehow, but i'm not sure.

Comment (resmo) [inline packaging/language/composer.py]: hmm I see. Do you have an example?

Generally we should return `changed=True` if we are not sure that something has changed, also for commands executed `changed=True` makes sense.

Comment (ramunasd) [inline packaging/language/composer.py]: After deeper investigation i see only 'composer init' for existing project returns empty string. So this check mainly for command 'init'.

I can remove this check if You see something bad.

Comment (resmo) [inline packaging/language/composer.py]: I would remove it. composer init would return changed=True which is what we want. So please remove it.

**Comment by gregdek** [comment]
@ramunasd is this now ready for review, in your opinion? If so,  please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!

**Comment by ramunasd** [comment]
@gregdek Hi, it's ready_for_review

**Comment by gregdek** [comment]
Thanks @ramunasd. 

@dmtrs @resmo Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

**Comment by resmo** [comment]
Ok, I think this change needs even to go further, because it actually fixes the command arguments handling. 

We expect the command to be a composer command but it is actually a string (so you are allowed by the module to add composer commands with arguemnts. e.g. 

```
- composer:
    command: "require my/package"
    working_dir: "/path/to/project"
```

But to find out all options available we did not expect this to be possible, it can cause this to fail. 

So we should add a check that command is a string not having a whitespace in it (e.g. no arguments). So the check for the options would only fail if the command is unknown, which is what we want.

So the example should be:

```
- composer:
   command: "require"
   arguments: " my/package"
   working_dir: "/path/to/project"
```

So in short:
Please extend this with a check to make sure, command has no whitespace (no arguments)

```
import re
if re.search(r"\s", command):
    module.fail_json(msg="Use the 'arguments' param for passing arguments with the 'command'")
```

**Comment by ramunasd** [comment]
Hi @resmo, check added.

File: packaging/language/composer.py [hunk 2]
```diff
@@ -102,6 +108,17 @@
 EXAMPLES = '''
 # Downloads and installs all the libs and dependencies outlined in the /path/to/project/composer.lock
 - composer: command=install working_dir=/path/to/project
+
+- composer:
+    command: "require my/package"
```
Comment (resmo) [inline packaging/language/composer.py]: Please fix this example with the new `arguments` param

**Comment by resmo** [comment]
shipit

**Comment by abadger** [comment]
Thanks @ramunasd and @resmo!
